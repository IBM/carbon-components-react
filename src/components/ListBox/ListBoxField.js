import cx from 'classnames';
import React from 'react';
import PropTypes from 'prop-types';

const ListBoxField = ({ children, ...rest }) => (
  <div className="bx--list-box__field" tabIndex="0" {...rest}>
    {children}
  </div>
);

ListBoxField.propTypes = {
  children: PropTypes.node,
};

export default ListBoxField;
