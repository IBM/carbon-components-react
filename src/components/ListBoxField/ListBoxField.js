import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import ListBoxLabel from '../ListBoxLabel';
import ListBoxMenuIcon from '../ListBoxMenuIcon';

const ListBoxField = ({
  type = 'default',
  label,
  selectedItem,
  clearSelection,
  isOpen,
  children,
  getButtonProps,
  // `rest` here should correspond with `getButtonProps` from `downshift`
  ...rest
}) => {
  const className = cx({
    'bx--list-box__field': true,
    'bx--list-box__field--inline': type === 'inline',
    'bx--list-box__field--open': isOpen,
  });
  return (
    <div className={className} tabIndex="0" {...rest}>
      {selectedItem && (
        <ListBoxLabel
          label={label}
          count={selectedItem.length}
          clearSelection={clearSelection}
        />
      )}
      {children}
      <ListBoxMenuIcon isOpen={isOpen} />
    </div>
  );
};

ListBoxField.propTypes = {
  type: PropTypes.oneOf(['default', 'inline']),
  label: PropTypes.string.isRequired,
  selectedItem: PropTypes.array,
  clearSelection: PropTypes.func,
  isOpen: PropTypes.bool.isRequired,
};

export default ListBoxField;
