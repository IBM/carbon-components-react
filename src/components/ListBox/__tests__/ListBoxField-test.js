import React from 'react';
import { mount } from 'enzyme';
import ListBoxField from '../ListBoxField';

describe('ListBoxField', () => {
  it('should render', () => {
    const wrapper = mount(
      <ListBoxField>
        <span>Hello</span>
      </ListBoxField>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should be focusable', () => {
    const wrapper = mount(
      <ListBoxField>
        <span>Hello</span>
      </ListBoxField>
    );
    expect(wrapper.children().prop('tabIndex')).toBe('0');
  });
});
