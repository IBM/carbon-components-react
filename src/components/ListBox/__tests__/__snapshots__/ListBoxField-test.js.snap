// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListBoxField should render 1`] = `
<ListBoxField>
  <div
    className="bx--list-box__field"
    tabIndex="0"
  >
    <span>
      Hello
    </span>
  </div>
</ListBoxField>
`;
